code/modules/power/apc/apc_main.dm

```diff
@@ -183,6 +181,10 @@
 
 	addtimer(CALLBACK(src, .proc/update), 5)
 
+	///This is how we test to ensure that mappers use the directional subtypes of APCs, rather than use the parent and pixel-shift it themselves.
+	if(abs(offset_old) != APC_PIXEL_OFFSET)
+		log_mapping("APC: ([src]) at [AREACOORD(src)] with dir ([dir] | [uppertext(dir2text(dir))]) has pixel_[dir & (WEST|EAST) ? "x" : "y"] value [offset_old] - should be [dir & (SOUTH|EAST) ? "-" : ""][APC_PIXEL_OFFSET]. Use the directional/ helpers!")
```

Comment:
~~we should make log_mapping cause ci to error prove me wrong~~

Reply:
They do fail! However, not every log type fails. Sometimes maps break the rules and we don't want CI to fail because of """intentional behavior""", as declared in https://github.com/tgstation/tgstation/blob/master/code/modules/unit_tests/mapping.dm :

```// Only fail if AREACOORD was conveyed, and it's a station or mining z-level.
// This is due to mapping errors don't have coords being impossible to diagnose as a unit test,
// and various ruins frequently intentionally doing non-standard things.```

This behavior was introduced in #66864 by the by.